frame/nfts/src/lib.rs

```diff
@@ -0,0 +1,1725 @@
+		/// - `witness_data`: When the mint type is `HolderOf(collection_id)`, then the owned
+		///   item_id from that collection needs to be provided within the witness data object.
+		///
+		/// Emits `Issued` event when successful.
+		///
+		/// Weight: `O(1)`
+		#[pallet::weight(T::WeightInfo::mint())]
+		pub fn mint(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			item: T::ItemId,
+			witness_data: Option<MintWitness<T::ItemId>>,
+		) -> DispatchResult {
+			let caller = ensure_signed(origin)?;
+
+			let collection_config = Self::get_collection_config(&collection)?;
+			let item_settings = collection_config.mint_settings.default_item_settings;
+			let item_config = ItemConfig { settings: item_settings };
+
+			Self::do_mint(
+				collection,
+				item,
+				caller.clone(),
+				item_config,
+				false,
+				|collection_details, collection_config| {
+					let mint_settings = collection_config.mint_settings;
+					let now = frame_system::Pallet::<T>::block_number();
+
+					if let Some(start_block) = mint_settings.start_block {
+						ensure!(start_block <= now, Error::<T, I>::MintNotStarted);
+					}
+					if let Some(end_block) = mint_settings.end_block {
+						ensure!(end_block >= now, Error::<T, I>::MintEnded);
+					}
+
+					match mint_settings.mint_type {
+						MintType::Issuer => {
+							ensure!(
+								Self::has_role(&collection, &caller, CollectionRole::Issuer),
+								Error::<T, I>::NoPermission
+							)
+						},
+						MintType::HolderOf(collection_id) => {
+							let MintWitness { owner_of_item } =
+								witness_data.ok_or(Error::<T, I>::BadWitness)?;
+
+							let has_item = Account::<T, I>::contains_key((
+								&caller,
+								&collection_id,
+								&owner_of_item,
+							));
+							ensure!(has_item, Error::<T, I>::BadWitness);
+
+							let attribute_key = Self::construct_attribute_key(
+								PalletAttributes::<T::CollectionId>::UsedToClaim(collection)
+									.encode(),
+							)?;
+
+							let key = (
+								&collection_id,
+								Some(owner_of_item),
+								AttributeNamespace::Pallet,
+								&attribute_key,
+							);
+							let already_claimed = Attribute::<T, I>::contains_key(key.clone());
+							ensure!(!already_claimed, Error::<T, I>::AlreadyClaimed);
+
+							let value = Self::construct_attribute_value(vec![0])?;
+							Attribute::<T, I>::insert(
+								key,
+								(value, AttributeDeposit { account: None, amount: Zero::zero() }),
+							);
+						},
+						_ => {},
+					}
+
+					if let Some(price) = mint_settings.price {
+						T::Currency::transfer(
+							&caller,
+							&collection_details.owner,
+							price,
+							ExistenceRequirement::KeepAlive,
+						)?;
+					}
+
+					Ok(())
+				},
+			)
+		}
+
+		/// Mint an item of a particular collection from a privileged origin.
+		///
+		/// The origin must conform to `ForceOrigin` or must be `Signed` and the sender must be the
+		/// Issuer of the `collection`.
```

Comment:
I would just make it `ForceOrigin` as the Issuer can just use `mint`.

Reply:
the `mint` method doesn't allow to specify the receiver of the NFT. Maybe we should rename this method from force_mint to smth else?

Reply:
What is the reason for that? The `mint` method in the Assets pallet allows the issuer to specify the account into which the assets are minted.

Reply:
Addressed here: https://github.com/paritytech/substrate/pull/12997

Reply:
p.s. the force_mint method additionally allows specifying the config of the new item while the default mint() method takes the `default_item_settings` from the `config.mint_settings`.
Usually, collection owners want to provide custom settings. That's why I'm suggesting allowing them to use the force_mint() method.